### PR TITLE
Creating signup feature

### LINKED ISSUE #85: Create Sign up feature
- [x] Add URL route.
- [x] Add signup form.
- [x] Update the signup URL on the homepage.
- [x] Build HTML template.
- [x] Add the view to views.py.
- [x] Set up css for sign-up template.
- [x] Add tests.
- [ ] Update the TODO's

### PR DESCRIPTION
**Pull request by rawadabu**
resolve #85 

**Sign Up Feature**

Added view and form for signup feature.
Added css file for signup UI.
Added an appropriate tests for form, view, valid and invalid inputs.

### CODE REVIEW

File: projboard/templates/base.html [hunk 1]
```diff
@@ -49,6 +49,30 @@
...
+        {% else %}
+
+        <ul class="nav__links">
+          <li class="nav__item">
+            <a class="nav__link nav__link--btn" href="/"
+              >Logout</a
+            >
+          </li>
+        </ul>
+        {% endif %}
```
Comment (finziamit) [inline projboard/templates/base.html:70]: Great Work!
Seems like Yael also modifies the navbar in base.html for authenticated users and guests in PR #90 so maybe it's unnecessary here

**Review by finziamit — COMMENTED** [review]
Great work!

File: projboard/templates/base.html (hunk 1 above)
Comment (Shalevro2) [reply projboard/templates/base.html:70]: In my opinion it's ok, because we can resolve it in the merge conflict.

Comment (rawadabu) [reply projboard/templates/base.html:70]: Yes I can resolve conflicts after merging, no problem

File: projboard/tests/test_signup.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        'name': VALID_NAME,
+        'nickname': VALID_USER,
+        'password': VALID_PASSWORD,
+        'password_confirm': VALID_PASSWORD,
+    }
+    form = NewUserForm(data=form_data)
+    if form.is_valid():
+        assert True
+    else:
+        assert False
```
Comment (Shalevro2) [inline projboard/tests/test_signup.py]: Can we write here just `assert form.is_valid()`?

File: projboard/tests/test_signup.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        'name': VALID_NAME,
+        'nickname': VALID_USER,
+        'password': VALID_PASSWORD,
+        'password_confirm': VALID_PASSWORD,
+    }
+    form = NewUserForm(data=form_data)
+    if form.is_valid():
+        assert False
+    else:
+        assert True
```
Comment (Shalevro2) [inline projboard/tests/test_signup.py]: Can we write here just `assert not form.is_valid()`?

**Review by Shalevro2 — APPROVED** [review]
LGTM

**Review by yaelnoyman1 — APPROVED** [review]
LGTM

**Review by finziamit — APPROVED** [review]
LGTM

File: projboard/tests/test_signup.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        'nickname': VALID_USER,
+        'password': VALID_PASSWORD,
+        'password_confirm': VALID_PASSWORD,
+        }
+    response = client.post(SIGNUP_PATH, data=valid_data)
+    # ASSERTION TO SHOW THAT SIGNUP SWITCHED TO HOMEPAGE (LOGIN PAGE LATER)
+    assert response.status_code == 302
+    assert User.get_user_by_nickname(VALID_USER)
+    # THIS ASSERT IS SETTED FOR HOMEPAGE BECAUSE /LOGIN ARENT READY YET
+    assert response.url == '/'
```
Comment (Yarboa) [inline projboard/tests/test_signup.py:56]: Correct, If you receive 302 
You could query again with client that you receive the correct page when you request

client.get("/")
And verify you are in the correct template

File: projboard/tests/test_signup.py (hunk 3 above)
Comment (rawadabu) [reply projboard/tests/test_signup.py]: Done

File: projboard/tests/test_signup.py (hunk 2 above)
Comment (rawadabu) [reply projboard/tests/test_signup.py]: Done

File: projboard/tests/test_signup.py (hunk 4 above)
Comment (rawadabu) [reply projboard/tests/test_signup.py:56]: @Yarboa After merging login feature I will change it `assert response.url == '/login'`, so after valid registration, user will be rendered to login page

Comment (Yarboa) [reply projboard/tests/test_signup.py:56]: You should verify the you continue to the right place after 302 reponse
It is another client client call

Comment (rawadabu) [reply projboard/tests/test_signup.py:56]: @Yarboa Got it, please check 01b8113

Comment (rawadabu) [reply projboard/tests/test_signup.py:56]: @Yarboa And rebased to #96

Comment (Yarboa) [reply projboard/tests/test_signup.py:56]: Almost there, please check the template received, assert you are in the right place

File: projboard/tests/test_signup.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+def test_invalid_password_registration(client):
+    invalid_data = {
+        'email': VALID_EMAIL,
+        'name': VALID_NAME,
+        'nickname': VALID_USER,
+        'password': VALID_PASSWORD,
+        'password_confirm': INVALID_PASSWORD_MISMATCH,
+        }
+    response = client.post(SIGNUP_PATH, data=invalid_data)
+    assert response.status_code == 200
```
Comment (Yarboa) [inline projboard/tests/test_signup.py:87]: Please assert the template returned

**Review by Yarboa — COMMENTED** [review]
@rawadabu almost there lets merge it in the next commit

File: projboard/tests/test_signup.py (hunk 5 above)
Comment (rawadabu) [reply projboard/tests/test_signup.py:87]: @Yarboa Please check [cda422f](https://github.com/redhat-beyond/Smarticle/pull/94/commits/cda422f4caa4ba1d7516715c3d3f20ff0b40a444)

File: projboard/tests/test_signup.py (hunk 4 above)
Comment (rawadabu) [reply projboard/tests/test_signup.py:56]: @Yarboa Done

**Review by Yarboa — APPROVED** [review]
LGTM

**Pull request merged**